From eae42c4c7789527f3871f9fceb953dd25b967087 Mon Sep 17 00:00:00 2001
From: "Dr. Stefan Schimanski" <stefan.schimanski@gmail.com>
Date: Mon, 17 Feb 2014 03:18:18 -0500
Subject: [PATCH 1/7] Backport of TLS 1.2 support to ACE 5.8.1

Original commit: svn://svn.dre.vanderbilt.edu/DOC/Middleware/trunk@96825
---
 ACE/ChangeLog                                      | 16 ++++++++++++++++
 ACE/THANKS                                         |  3 +++
 ACE/ace/SSL/SSL_Context.cpp                        | 22 ++++++++++++++++++++++
 ACE/ace/SSL/SSL_Context.h                          |  6 ++++++
 ACE/examples/C++NPv2/AIO_Client_Logging_Daemon.cpp |  6 +++---
 5 files changed, 50 insertions(+), 3 deletions(-)

diff --git a/ACE/ChangeLog b/ACE/ChangeLog
index 6bff2f1..acbcf10 100644
--- a/ACE/ChangeLog
+++ b/ACE/ChangeLog
@@ -1,3 +1,19 @@
+Sat Feb 16 14:42:54 UTC 2013  Douglas C. Schmidt  <schmidt@dre.vanderbilt.edu>
+
+        * examples/C++NPv2/AIO_Client_Logging_Daemon.cpp: Moved the definition of
+  
+          ACE_FACTORY_DEFINE (AIO_CLD, AIO_Client_Logging_Daemon)
+        
+          to the right place in the #ifdefs.  Thanks to Journeyer J. Joh
+          <oosaprogrammer at gmail dot com> for reporting this.
+
+Sat Feb 16 14:34:30 UTC 2013  Douglas C. Schmidt  <schmidt@dre.vanderbilt.edu>
+
+        * ace/SSL/SSL_Context.cpp (ACE_SSL_Context::set_mode),
+        * ace/SSL/SSL_Context.h (ACE_SSL_Context): Added support for TLS
+          1.1 and 1.2.  Thanks to pkow88776 at onet dot pl for
+          contributing this.
+
 Sun Aug 01 11:30:54 CEST 2010  Johnny Willemsen  <jwillemsen@remedy.nl>
 
 	* ACE version 5.8.1 released.
diff --git a/ACE/THANKS b/ACE/THANKS
index a96ae85..020cbe8 100644
--- a/ACE/THANKS
+++ b/ACE/THANKS
@@ -2328,6 +2328,9 @@ David Ward <dpward at mit dot edu>
 Anatoli Sakhnik <sakhnik at gmail dot com>
 Max Zhou <earthdog at 126 dot com>
 Daynesh Mangal <daynesh at gmail dot com>
+Phillip LaBanca <labancap at ociweb dot com>
+<pkow88776 at onet dot pl>
+Journeyer J. Joh <oosaprogrammer at gmail dot com>
 
 I would particularly like to thank Paul Stephenson, who worked with me
 at Ericsson in the early 1990's.  Paul devised the recursive Makefile
diff --git a/ACE/ace/SSL/SSL_Context.cpp b/ACE/ace/SSL/SSL_Context.cpp
index 2512e71..906bac6 100644
--- a/ACE/ace/SSL/SSL_Context.cpp
+++ b/ACE/ace/SSL/SSL_Context.cpp
@@ -280,6 +280,28 @@ ACE_SSL_Context::set_mode (int mode)
     case ACE_SSL_Context::TLSv1:
       method = ::TLSv1_method ();
       break;
+#ifdef TLS1_1_VERSION
+    case ACE_SSL_Context::TLSv1_1_client:
+      method =3D ::TLSv1_1_client_method ();
+      break;
+    case ACE_SSL_Context::TLSv1_1_server:
+      method =3D ::TLSv1_1_server_method ();
+      break;
+    case ACE_SSL_Context::TLSv1_1:
+      method =3D ::TLSv1_1_method ();
+      break;
+#endif
+#ifdef TLS1_2_VERSION
+    case ACE_SSL_Context::TLSv1_2_client:
+      method =3D ::TLSv1_2_client_method ();
+      break;
+    case ACE_SSL_Context::TLSv1_2_server:
+      method =3D ::TLSv1_2_server_method ();
+      break;
+    case ACE_SSL_Context::TLSv1_2:
+      method =3D ::TLSv1_2_method ();
+      break;
+#endif
     default:
       method = ::SSLv3_method ();
       break;
diff --git a/ACE/ace/SSL/SSL_Context.h b/ACE/ace/SSL/SSL_Context.h
index 575e5ca..07e4abd 100644
--- a/ACE/ace/SSL/SSL_Context.h
+++ b/ACE/ace/SSL/SSL_Context.h
@@ -94,6 +94,12 @@ public:
     TLSv1_client,
     TLSv1_server,
     TLSv1
+    TLSv1_1_client,
+    TLSv1_1_server,
+    TLSv1_1,
+    TLSv1_2_client,
+    TLSv1_2_server,
+    TLSv1_2
   };
 
   /// Constructor
diff --git a/ACE/examples/C++NPv2/AIO_Client_Logging_Daemon.cpp b/ACE/examples/C++NPv2/AIO_Client_Logging_Daemon.cpp
index 3d7e36c..618f979 100644
--- a/ACE/examples/C++NPv2/AIO_Client_Logging_Daemon.cpp
+++ b/ACE/examples/C++NPv2/AIO_Client_Logging_Daemon.cpp
@@ -32,8 +32,6 @@
 #include "AIO_Client_Logging_Daemon.h"
 #include <openssl/ssl.h>
 
-
-
 class AIO_CLD_Acceptor
   : public ACE_Asynch_Acceptor<AIO_Input_Handler> {
 public:
@@ -357,6 +355,8 @@ int AIO_Client_Logging_Daemon::fini () {
   return 0;
 }
 
+#endif /* (ACE_WIN32 && !ACE_HAS_WINCE) || ACE_HAS_AIO_CALLS */
+
 ACE_FACTORY_DEFINE (AIO_CLD, AIO_Client_Logging_Daemon)
 
-#endif /* (ACE_WIN32 && !ACE_HAS_WINCE) || ACE_HAS_AIO_CALLS */
+
-- 
1.7.12.4

